autoscale-groups-nodes-plugin/autoscale-groups-nodes-plugin-core/src/main/java/fish/payara/extensions/autoscale/groups/nodes/admin/GetNodesScalingGroupConfigurationCommand.java

```diff
@@ -0,0 +1,139 @@
+                params = {
+                        @RestParam(name = "id", value = "$parent")
+                }
+        )
+})
+public class GetNodesScalingGroupConfigurationCommand extends GetScalingGroupConfigurationCommand {
+
+    @Override
+    public void execute(AdminCommandContext adminCommandContext) {
+        try {
+            validateParams();
+        } catch (CommandValidationException commandValidationException) {
+            adminCommandContext.getActionReport().setFailureCause(commandValidationException);
+            adminCommandContext.getActionReport().setActionExitCode(ActionReport.ExitCode.FAILURE);
+            return;
+        }
+
+        NodesScalingGroup nodesScalingGroup = null;
+
+        // Search through the scaling groups, checking for the one with a reference to our requested deployment group
+        for (NodesScalingGroup nodesScalingGroupIterator : scalingGroups.getScalingGroupsOfType(NodesScalingGroup.class)) {
+            if (nodesScalingGroupIterator.getName().equals(name)) {
+                nodesScalingGroup = nodesScalingGroupIterator;
+                break;
+            }
+        }
+
+        adminCommandContext.getActionReport().setMessage("Nodes Scaling Group: " + nodesScalingGroup.getName());
+        adminCommandContext.getActionReport().appendMessage("\nConfig Ref: " + nodesScalingGroup.getConfigRef());
+        adminCommandContext.getActionReport().appendMessage("\nDeployment Group Ref: " + nodesScalingGroup.getDeploymentGroupRef());
+        adminCommandContext.getActionReport().appendMessage("\nNode Refs: " + String.join(", ", nodesScalingGroup.getNodeRefs()));
+
+        Properties extraProps = new Properties();
+        Map<String, Object> configMap = new HashMap<>();
+        configMap.put("name", nodesScalingGroup.getName());
+        configMap.put("config", nodesScalingGroup.getConfigRef());
+        configMap.put("deploymentGroup", nodesScalingGroup.getDeploymentGroupRef());
+        configMap.put("nodes", nodesScalingGroup.getNodeRefs());
+
+        extraProps.put("scalingGroupConfig", configMap);
+        adminCommandContext.getActionReport().setExtraProperties(extraProps);
+    }
+
+    @Override
+    protected void validateParams() throws CommandValidationException {
+        super.validateParams();
+
+        List<NodesScalingGroup> nodesScalingGroups = scalingGroups.getScalingGroupsOfType(NodesScalingGroup.class);
+
+        if (nodesScalingGroups.isEmpty()) {
+            throw new CommandValidationException("Scaling Group " + name + " is not a Nodes Scaling Group.");
+        }
+
+        boolean exists = false;
+        for (NodesScalingGroup nodesScalingGroup : nodesScalingGroups) {
+            if (nodesScalingGroup.getName().equals(name)) {
+                exists = true;
+                break;
+            }
+        }
+
+        if (!exists) {
+            throw new CommandValidationException("Scaling Group " + name + " is not a Nodes Scaling Group.");
+        }
```

Comment:
Can be easily replaced with a lambda expression to check if the group exists, for better readibility

Reply:
Writer preference, it's clear enough I think.